pmd-java/src/main/java/net/sourceforge/pmd/lang/java/rule/bestpractices/UnusedPrivateFieldRule.java

```diff
@@ -75,14 +75,13 @@ private boolean isOK(ASTFieldDeclaration field) {
     }
 
     private boolean hasAnyAnnotation(Annotatable node) {
-        NodeStream<ASTAnnotation> declaredAnnotations = node.getDeclaredAnnotations();
         for (String reportAnnotation : getProperty(REPORT_FOR_ANNOTATIONS_DESCRIPTOR)) {
-            for (ASTAnnotation annotation: declaredAnnotations) {
```

Comment:
It's either-or, so the redundancy in the code `node.getDeclaredAnnotations()` might be acceptable in favor of resolving the field. The getter is called only once either way. This way, we save the interpreter from declaring the field.  

This should arguably make the code smaller but maybe faster, even if just a little bit.

Reply:
It is not called once anyway. This line is in a loop so by inlining this variable the getter will be called many times in the enclosing loop.

Also consider that this is not a field, it's a variable, and there is no runtime overhead to declaring a variable.